src/test/java/htsjdk/samtools/cram/encoding/external/ExternalCodecEquivalenceTest.java

```diff
@@ -0,0 +1,176 @@
+        try (final ByteArrayOutputStream os = new ByteArrayOutputStream()) {
+            final CRAMCodec<Byte> writeCodec = new ExternalByteCodec(null, os);
+
+            for (final T value : values) {
+                writeCodec.write(value.byteValue());
+            }
+            os.flush();
+            written = os.toByteArray();
+        } catch (final IOException e) {
+            throw new RuntimeIOException(e);
+        }
+        return written;
+    }
+
+    private <T extends Number> byte[] externalIntegerCodecWrite(List<T> values) {
+        byte[] written;
+        try (final ByteArrayOutputStream os = new ByteArrayOutputStream()) {
+            final CRAMCodec<Integer> writeCodec = new ExternalIntegerCodec(null, os);
+
+            for (final T value : values) {
+                writeCodec.write(value.intValue());
+            }
+            os.flush();
+            written = os.toByteArray();
+        } catch (final IOException e) {
+            throw new RuntimeIOException(e);
+        }
+        return written;
+    }
+
+
+    private <T extends Number> byte[] externalLongCodecWrite(List<T> values) {
+        byte[] written;
+        try (final ByteArrayOutputStream os = new ByteArrayOutputStream()) {
+            final CRAMCodec<Long> writeCodec = new ExternalLongCodec(null, os);
+
+            for (final T value : values) {
+                writeCodec.write(value.longValue());
+            }
+            os.flush();
+            written = os.toByteArray();
+        } catch (final IOException e) {
+            throw new RuntimeIOException(e);
+        }
+        return written;
+    }
+
+    private List<Long> externalByteCodecRead(byte[] written, final int length) {
+        return read(written, length, this::externalByteCodecReaderConstructor);
+    }
+
+    private List<Long> externalIntegerCodecRead(byte[] written, final int length) {
+        return read(written, length, this::externalIntegerCodecReaderConstructor);
+    }
+
+    private List<Long> externalLongCodecRead(byte[] written, final int length) {
+        return read(written, length, this::externalLongCodecReaderConstructor);
+    }
+
+    private <T extends Number> List<Long> read(byte[] written, final int length, final ReadCodecConstructor<T> readCC) {
+        List<Long> retval = new ArrayList<>();
+
+        try (final ByteArrayInputStream is = new ByteArrayInputStream(written)) {
+            final CRAMCodec<T> readCodec = readCC.reader(is);
+            for (int i = 0; i < length; i++) {
+                retval.add(readCodec.read().longValue());
+            }
+        } catch (final IOException e) {
+            throw new RuntimeIOException(e);
+        }
+        return retval;
+    }
+
+    // TODO: move these into the Codec classes?
+
+    private interface ReadCodecConstructor<T> {
+        CRAMCodec<T> reader(final ByteArrayInputStream is);
+    }
+
+    private CRAMCodec<Byte> externalByteCodecReaderConstructor(final ByteArrayInputStream is) {
```

Comment:
These methods seem like good candidates for inlining?  Did you want to add an interface method to access these instead somehow? The todo makes it sound that way.

Reply:
like this?  
`codecPairTest(values, this::externalIntegerCodecWrite,  ((bytes, len) -> read(bytes, len, this::externalByteCodecReaderConstructor)));`

Reply:
Oh, bleh, I think I misread where these were used.

Reply:
That looks less readable to me, and I don't think it matters much for a small test like this

Reply:
No.. these are used only in the 1 method above them:
```
    private List<Long> externalLongCodecRead(byte[] written, final int length) {
        return read(written, length, is -> new ExternalLongCodec(is, null));
    }
```

It would stay `codecPairTest(values, this::externalIntegerCodecWrite,  this::externalByteCodecRead); `

Reply:
A lot of these methods can be made static too while we're here.

Reply:
If you think it's less readable then leave it how it is.

Reply:
oh I see - yes I'll make these inline.

What is the benefit of static for test classes?  I have not generally been doing that.